### DISCUSSION TITLE
Xcode provided project for c++

### DISCUSSION THREAD

**Discussion by amritpan** [antlr/antlr4#3873] [General]
Hi all,

I'm looking to set up ANTLR4 on Xcode for C++ and a saw a reference to a provided project for Xcode on the C++ target doc but there is no link to it.  

["Instead of downloading a prebuilt binary you can also easily build your own library on OSX or Windows. Just use the provided projects for XCode or Visual Studio and build it."](https://github.com/antlr/antlr4/blob/master/doc/cpp-target.md) Where can I download/clone this?

**Comment by parrt** [comment]
I think if you look under the runtime/Cpp you will find something interesting.

**Comment by chris-miner** [comment]
There are a couple of Xcode project files:
./runtime/Cpp/demo/Mac/antlrcpp-demo.xcodeproj
./runtime/Cpp/runtime/antlrcpp.xcodeproj

Neither of them result in a functional build experience out of the box.  Could be something simple.  The runtime project complains of missing files.  For example, StarBlockState.cpp and TokensStartState.cpp are referenced but non-existent.  Maybe they are supposed to be generated?  Oddly enough there is a TokensStartState for c#, swift, java, and javascript

The cpp version was deleted in commit commit 61ca09ec27c80cf7e94f7d709c69e5dc28097de4 on Sun Apr 10 at 09:55:13 2022 -0700

So ... maybe it is enough to remove that from the Xcode project?  Could be the same deal for the other files.  Seems odd though that they are in other targets and the header is still around.
